mmdeploy/codebase/mmocr/models/text_detection/panet_trt_fp16.py

```diff
@@ -0,0 +1,135 @@
+        self: The instance of the class FPEM_FFM.
+        x (List[Tensor]): A list of feature maps of shape (N, C, H, W).
+
+    Returns:
+        outs (List[Tensor]): A list of feature maps of shape (N, C, H, W).
+    """
+    c2, c3, c4, c5 = x
+    # reduce channel
+    c2 = self.reduce_conv_c2(c2)
+    c3 = self.reduce_conv_c3(c3)
+    c4 = self.reduce_conv_c4(c4)
+
+    if ENABLE:
+        bn_w = self.reduce_conv_c5[1].weight / torch.sqrt(
+            self.reduce_conv_c5[1].running_var + self.reduce_conv_c5[1].eps)
+        bn_b = self.reduce_conv_c5[
+            1].bias - self.reduce_conv_c5[1].running_mean * bn_w
+        bn_w = bn_w.reshape(1, -1, 1, 1).repeat(1, 1, c5.size(2), c5.size(3))
+        bn_b = bn_b.reshape(1, -1, 1, 1).repeat(1, 1, c5.size(2), c5.size(3))
+        conv_b = self.reduce_conv_c5[0].bias.reshape(1, -1, 1, 1).repeat(
+            1, 1, c5.size(2), c5.size(3))
+        c5 = FACTOR * (self.reduce_conv_c5[:-1](c5)) - (FACTOR - 1) * (
+            bn_w * conv_b + bn_b)
+        c5 = self.reduce_conv_c5[-1](c5)
+    else:
+        c5 = self.reduce_conv_c5(c5)
+
+    # FPEM
+    for i, fpem in enumerate(self.fpems):
+        c2, c3, c4, c5 = fpem(c2, c3, c4, c5)
+        if i == 0:
+            c2_ffm = c2
+            c3_ffm = c3
+            c4_ffm = c4
+            c5_ffm = c5
+        else:
+            c2_ffm += c2
+            c3_ffm += c3
+            c4_ffm += c4
+            c5_ffm += c5
+
+    # FFM
+    c5 = F.interpolate(
+        c5_ffm,
+        c2_ffm.size()[-2:],
+        mode='bilinear',
+        align_corners=self.align_corners)
+    c4 = F.interpolate(
+        c4_ffm,
+        c2_ffm.size()[-2:],
+        mode='bilinear',
+        align_corners=self.align_corners)
+    c3 = F.interpolate(
+        c3_ffm,
+        c2_ffm.size()[-2:],
+        mode='bilinear',
+        align_corners=self.align_corners)
+    outs = [c2_ffm, c3, c4, c5]
+    return tuple(outs)
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmdet.models.backbones.resnet.BasicBlock.forward',
+    backend=Backend.TENSORRT.value)
+def basic_block__forward__trt(ctx, self, x: torch.Tensor) -> torch.Tensor:
+    """Rewrite `forward` of BasicBlock for tensorrt backend.
+
+    Rewrite this function avoid overflow for tensorrt-fp16 with the checkpoint
+    `https://download.openmmlab.com/mmocr/textdet/panet/panet_r18_fpem_ffm
+    _sbn_600e_icdar2015_20210219-42dbe46a.pth`
+
+    Args:
+        ctx (ContextCaller): The context with additional information.
+        self: The instance of the class FPEM_FFM.
+        x (Tensor): The input tensor of shape (N, C, H, W).
+
+    Returns:
+        outs (Tensor): The output tensor of shape (N, C, H, W).
+    """
+    if self.conv1.in_channels < CHANNEL_THRESH:
+        return ctx.origin_func(self, x)
+
+    identity = x
+
+    out = self.conv1(x)
+    out = self.norm1(out)
+    out = self.relu(out)
+
+    out = self.conv2(out)
+
+    if torch.abs(self.norm2(out)).max() < 65504:
```

Comment:
Note that this is the result of the example input. The result model might still overflow on real data.